src/notation/view/pianokeyboard/pianokeyboardcontroller.cpp

```diff
@@ -87,6 +87,24 @@ void PianoKeyboardController::setPressedKey(std::optional<piano_key_t> key)
     m_keyStatesChanged.notify();
 }
 
+void PianoKeyboardController::setHoveredKey(std::optional<piano_key_t> key)
+{
+    auto notation = currentNotation();
```

Comment:
I'd move this after the `m_hoveredKey == key` check, to avoid querying the current notation unnecessarily